### PR TITLE
mds: set hb to NULL after the memory released

### PR DESCRIPTION
**Pull request by lxbsz**
Somewhere will test the `hb`, it seems possible that this could cause use after free bug.

Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by batrick — APPROVED** [review]

File: src/mds/MDSRank.cc [hunk 1]
```diff
@@ -556,6 +556,7 @@ MDSRank::~MDSRank()
 {
   if (hb) {
     g_ceph_context->get_heartbeat_map()->remove_worker(hb);
+    hb = NULL;
```
Comment (batrick) [inline src/mds/MDSRank.cc:559]:
```suggestion
    hb = nullptr;
```

would be idiomatic c++

Comment (vshankar) [reply src/mds/MDSRank.cc:559]: ... and there is another `hb = NULL` in `shutdown()`.

Also, I think by the time we hit the destructor `hb` is already nullptr, but its better to be safe.

Comment (lxbsz) [reply src/mds/MDSRank.cc]: > would be idiomatic c++
Done.

Comment (lxbsz) [reply src/mds/MDSRank.cc]: > ... and there is another `hb = NULL` in `shutdown()`.
> 
> Also, I think by the time we hit the destructor `hb` is already nullptr, but its better to be safe.

Yeah, agree. I am not sure whether will there have some cases that the `mds_rank` hasn't initialized yet and directly call the `~MDSRank()`.

Comment (vshankar) [reply src/mds/MDSRank.cc]: hmmm..

Let's change to nullptr. This looks fine!

**Review by vshankar — APPROVED** [review]

**Pull request merged**